test/utils.py

```diff
@@ -47,6 +49,18 @@ def cibuildwheel_get_build_identifiers(project_path, env=None, *, prerelease_pyt
     return cmd_output.strip().split("\n")
 
 
+def _update_pip_cache_dir(env: dict[str, str]) -> None:
+    if platform == "linux":
```

Comment:
```suggestion
    # Fix for pip concurrency bug https://github.com/pypa/pip/issues/11340
    # See https://github.com/pypa/cibuildwheel/issues/1254 for discussion.
    # We should be able to remove this once we drop Python 3.6.

    if platform == "linux":
```

Reply:
If we add `# We should be able to remove this once we drop Python 3.6.` then we need to modify `test_dependency_versions.py` now (we will forget later).
I'm not even sure it's a good idea to drop this ever. We might be in a similar situation every time pip cache breaks between version (& they'll most likely drop python 3.7 soon after EOL in 8 months so we'll be in the same situation).

Reply:
> I'm not even sure it's a good idea to drop this ever. We might be in a similar situation every time pip cache breaks between version (& they'll most likely drop python 3.7 soon after EOL in 8 months so we'll be in the same situation).

Yeah, maybe. I'm not sure what the root cause is yet. I'll keep an eye on the pip issue and if we get some assurances maybe it can be reverted at some point. Generally I'd like to keep the test environment as close as possible to user environments. 

Anyway I'm happy, let's get this merged and maybe our CI will mean something again :)